### ISSUE TITLE
Pre-Fill Name For CC Payments

### ISSUE THREAD

**Issue by sprankhub** [PAYONE-GmbH/magento-1#277]
Could you pre-fill the name fields for CC payments? You should be able to retrieve the name from the address IMHO. This would improve the UX.

**Comment by sprankhub** [comment]
Why did you assign me here, @fjbender?

**Comment by fjbender** [comment]
My mistake, sorry. @hreinberger is on it.

**Comment by hreinberger** [comment]
So here's the thing about the cardholder name:
* our API doesn't require it
* our magento 1 extension doesn't even send this data to our API
* There's no way for us to validate if a name and cardpan match
* the cardholder can differ from the billing name (business cards, card of your spouse etc.)

So I kinda want to get of rid of these fields to streamline the checkout instead of storing data in them that can't be validated and can be as fake as the billing name. At the very least we could make them optional. We'll discuss this internally and will follow up on this shortly.

Thanks (as always) for your contribution so far!!

**Comment by sprankhub** [comment]
Oh wow, interesting news :-) I think not asking for the name at all would be best for the customer. If you do not need and use it anyway, there is no reason to ask for it.

So if the field is not required and used at all, we should be able to simply hide the fields until you implemented this in your module, right?

**Comment by borishaefele** [comment]
that would be extremely helpful from a customer perspective!

**Comment by hreinberger** [comment]
@sprankhub I haven't really tested what happens when you hide these fields. Maybe the extension is wired up in a way that this would break the payment. But from the viewpoint of our API there's no need to submit cardholder data to complete a payment.

**Comment by sprankhub** [comment]
@hreinberger status? Are you currently working on removing the name fields altogether?

**Comment by sprankhub** [comment]
Any update @hreinberger @fjbender?

**Comment by hreinberger** [comment]
Hi @sprankhub sorry this had to take the back seat due to other topics for the moment but we're still on it!

**Comment by sprankhub** [comment]
Sorry for bugging again, but any update @hreinberger? It should not be too hard if you really remove the fields :-) Thanks!

**Comment by hreinberger** [comment]
Hi @sprankhub 
turns out it's not that easy to find reliable info on whether or not a cardholder is required. While it's still true that this info is completely optional at the time of an authorisation, the cardholder name can get requested when there's a chargeback request by the issuer due to suspected fraud. 

However, the scheme rules https://www.mastercard.us/content/dam/mccom/global/documents/mastercard-rules.pdf don't state that card holder data must be acquired but rather may be requested. (chapter 5.10.4 - yes you have to read through that on your first day as a BS PAYONE employee and yes there's a test 🧐) Also I'm not sure how useful this info would be in a case of fraud where every piece of personal information is likely to be false.

So in conclusion: The cardholder data can be requested by the issuer in a chargeback situation, but the informational value would be questionable at least. The lazy solution would be to make the cardholder field configurable, though I'm not keen on adding another checkbox when we can find a reasonable default. @fjbender let's address this tomorrow.

**Comment by fjbender** [comment]
@hreinberger Can you check this with the Scheme Compliance Office?

**Comment by sprankhub** [comment]
Thanks for the detailed update! Looking forward to more updates :)

**Comment by sprankhub** [comment]
Seems to be implemented via #338.

**Comment by fjbender** [comment]
ACK, referenced it in the PR accordingly.
